flickit-assessment-kit/src/main/java/org/flickit/assessment/kit/adapter/out/persistence/kitversion/KitVersionMapper.java

```diff
@@ -14,11 +14,13 @@
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class KitVersionMapper {
 
-    public static KitVersionJpaEntity createParamToJpaEntity(AssessmentKitJpaEntity kit, CreateKitVersionPort.Param param) {
+    public static KitVersionJpaEntity createParamToJpaEntity(Long id, AssessmentKitJpaEntity kit, CreateKitVersionPort.Param param) {
+        var statusVersion = param.status().getId() == 2 ? -id : param.status().getId();
```

Comment:
**Suggestion:** Refactor the logic for setting statusVersion in createParamToJpaEntity to handle more generic cases and ensure that the logic is clear and understandable, avoiding magic numbers like '2' and '-id'. [general, importance: 7]
```suggestion
        var statusVersion = calculateStatusVersion(param.status().getId(), id);
```